lib/modules/hosts/youtube.js

```diff
@@ -5,7 +5,7 @@ export default {
 	name: 'youtube',
 	attribution: false,
 	domains: ['youtube.com', 'youtu.be'],
-	detect: ({ href }) => (/^https?:\/\/(?:youtu\.be|(?:www\.|m\.)?youtube\.com)\/?(?:watch|embed)?(?:.*?v=|v\/|\/)([\w-]+)/i).exec(href),
+	detect: ({ href }) => (/^https?:\/\/(?:www\.|m\.)?(?:youtu\.be\/|youtube\.com\/?(?:watch|embed)(?:.*?v=|v\/|\/))([\w-]+)/i).exec(href),
```

Comment:
I also came up with this method:

```
const videoIdRe = /^([\w-]+)/;

[...]
	detect: ({ href }) => {
		let url = new URL(href);
		return url.hostname.endsWith('youtu.be') ?
			videoIdRe.exec(url.pathname.substring(1)) :
			url.searchParams.has("v") ?
				videoIdRe.exec(url.searchParams.get("v")) :
				videoIdRe.exec(url.pathname.split("/")[2]);
```

It works with every URL I've found. (including /v/)

I feel like this is a lot more understandable than the regex. Whatever you think is better I can implement.